seed/seed20.go

```diff
@@ -370,6 +376,44 @@ func (s *seed20) addModelSnap(modelSnap *asserts.ModelSnap, essential bool, tm t
 }
 
 func (s *seed20) LoadMeta(tm timings.Measurer) error {
+	if err := s.loadModelMeta(nil, tm); err != nil {
+		return err
+	}
+
+	// extra snaps
+	runMode := []string{"run"}
+	for {
+		optSnap, done := s.nextOptSnap(nil)
+		if done {
+			break
+		}
+
+		_, err := s.addSnap(optSnap, optSnap, runMode, "latest/stable", "snaps", tm)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *seed20) LoadEssentialMeta(essentialTypes []snap.Type, tm timings.Measurer) error {
+	filterEssential := essentialSnapTypesToModelFilter(essentialTypes)
+	expectedEssentialSnapsNum := len(essentialTypes)
```

Comment:
Could this be moved down close to where it's used?

Reply:
you mean the *Num right?

Reply:
Yes